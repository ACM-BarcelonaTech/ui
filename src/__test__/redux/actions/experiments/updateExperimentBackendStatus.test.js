import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import initialExperimentState, { experimentTemplate } from '../../../../redux/reducers/experiments/initialState';
// import { initialPipelineState }
// from '../../../../redux/reducers/experimentSettings/initialState';

jest.mock('localforage');

const mockStore = configureStore([thunk]);

describe('updateExperimentBackendStatus', () => {
  const experimentId = 'experiment-1';

  const mockExperiment = {
    ...experimentTemplate,
    name: 'experiment-1',
    id: experimentId,
  };

  // const mockBackendStatus = {
  //   pipeline: {
  //     ...initialPipelineState,
  //     startDate: '2021-01-01T00:00:00.000Z',
  //     stopDate: '2021-01-01T10:00:00.000Z',
  //     status: 'SUCCEEDED',
  //   },
  //   gem2s: {
  //     ...initialPipelineState,
  //     startDate: '2021-01-01T00:00:00.000Z',
  //     stopDate: '2021-01-01T10:00:00.000Z',
  //     status: 'SUCCEEDED',
  //   },
  // };

  const mockState = {
    experiments: {
      ...initialExperimentState,
      [experimentId]: mockExperiment,
    },
  };

  it('Dispatches action when called', async () => {
    const store = mockStore(mockState);

    // await store.dispatch(updateExperimentBackendStatus(experimentId, mockBackendStatus));

    const actions = store.getActions();

    // expect(actions[0].type).toEqual(EXPERIMENTS_BACKEND_STATUS_UPDATED);

    expect(actions).toMatchSnapshot();
  });
});
