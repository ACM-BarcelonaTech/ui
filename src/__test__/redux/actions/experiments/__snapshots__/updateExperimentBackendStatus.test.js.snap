// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateExperimentBackendStatus Dispatches action when called 1`] = `
Array [
  Object {
    "payload": Object {
      "backendStatus": Object {
        "gem2s": Object {
          "completedSteps": Array [],
          "endDate": null,
          "startDate": "2021-01-01T00:00:00.000Z",
          "status": "SUCCEEDED",
          "stopDate": "2021-01-01T10:00:00.000Z",
        },
        "pipeline": Object {
          "completedSteps": Array [],
          "endDate": null,
          "startDate": "2021-01-01T00:00:00.000Z",
          "status": "SUCCEEDED",
          "stopDate": "2021-01-01T10:00:00.000Z",
        },
      },
      "experimentId": "experiment-1",
    },
    "type": "experiments/updateExperimentBackendStatus",
  },
]
`;
